test/integration/scheduler/framework_test.go

```diff
@@ -1037,124 +1037,422 @@ func TestPrebindPlugin(t *testing.T) {
+			testCtx := initTestSchedulerForFrameworkTest(
+				t,
+				testutils.InitTestAPIServer(t, "unreserve-reserve-plugin", nil),
+				2,
+				scheduler.WithProfiles(prof),
+				scheduler.WithFrameworkOutOfTreeRegistry(registry))
+			defer testutils.CleanupTest(t, testCtx)
+
+			// Create a best effort pod.
+			podName := fmt.Sprintf("%s-%d", "test-pod", index)
+			pod, err := createPausePod(testCtx.ClientSet,
+				initPausePod(&pausePodConfig{Name: podName, Namespace: testCtx.NS.Name}))
+			if err != nil {
+				t.Errorf("Error while creating a test pod: %v", err)
+			}
+
+			if test.fail {
+				if err = wait.Poll(10*time.Millisecond, 30*time.Second, podSchedulingError(testCtx.ClientSet, pod.Namespace, pod.Name)); err != nil {
+					t.Errorf("Expected a reasons other than Unschedulable, but got: %v", err)
+				}
+
+				for i, pl := range test.plugins {
+					if i <= test.failPluginIdx {
+						if pl.numReserveCalled != 1 {
+							t.Errorf("Reserve Plugins %s numReserveCalled = %d, want 1.", pl.name, pl.numReserveCalled)
+						}
+					} else {
+						if pl.numReserveCalled != 0 {
+							t.Errorf("Reserve Plugins %s numReserveCalled = %d, want 0.", pl.name, pl.numReserveCalled)
+						}
+					}
+
+					if pl.numUnreserveCalled != 1 {
+						t.Errorf("Reserve Plugin %s numUnreserveCalled = %d, want 1.", pl.name, pl.numUnreserveCalled)
+					}
+				}
+			} else {
+				if err = testutils.WaitForPodToSchedule(testCtx.ClientSet, pod); err != nil {
+					t.Errorf("Expected the pod to be scheduled. error: %v", err)
+				}
+
+				for _, pl := range test.plugins {
+					if pl.numReserveCalled != 1 {
+						t.Errorf("Reserve Plugin %s numReserveCalled = %d, want 1.", pl.name, pl.numReserveCalled)
+					}
+					if pl.numUnreserveCalled != 0 {
+						t.Errorf("Reserve Plugin %s numUnreserveCalled = %d, want 0.", pl.name, pl.numUnreserveCalled)
+					}
+				}
+			}
+			testutils.CleanupPods(testCtx.ClientSet, t, []*v1.Pod{pod})
+		})
+	}
+}
+
+// TestUnReservePermitPlugins tests unreserve of Permit plugins.
+func TestUnReservePermitPlugins(t *testing.T) {
+	tests := []struct {
+		name   string
+		plugin *PermitPlugin
+		fail   bool
+	}{
+		{
+			name: "All Reverse plugins passed, but a Permit plugin was rejected",
+			fail: true,
+			plugin: &PermitPlugin{
+				name:         "rejectedPermitPlugin",
+				rejectPermit: true,
+			},
+		},
+		{
+			name: "All Reverse plugins passed, but a Permit plugin timeout in waiting",
+			fail: true,
+			plugin: &PermitPlugin{
+				name:          "timeoutPermitPlugin",
+				timeoutPermit: true,
+			},
+		},
+		{
+			name: "The Permit plugin succeed",
+			fail: false,
+			plugin: &PermitPlugin{
+				name: "succeedPermitPlugin",
+			},
+		},
+	}
+
+	for index, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			reservePlugins := []*ReservePlugin{
+				{
+					name:        "reversePlugin-1",
+					failReserve: false,
+				},
+				{
+					name:        "reversePlugin-2",
+					failReserve: false,
+				},
```

Comment:
typo: reverse -> reserve

btw: any particular reason to have 2 reserve plugins instead of one?